src/crypto/local.js

```diff
@@ -307,51 +308,71 @@ function verifyProof(proofObj, tokens, signatures, commitments) {
     let cZ = new sjcl.ecc.pointJac(CURVE);
+    const prng = {name: prngName};
+    switch (prng.name) {
+        case "shake":
+            prng.func = createShake256();
+            prng.func.update(seed, "hex");
+            break;
+        case "hkdf":
+            prng.func = evaluateHkdf;
+            break;
+        default:
+            throw new Error(`Server specified PRNG is not compatible: ${prng.name}`);
+    }
+    let iter = -1;
     for (let i=0; i<chkM.length; i++) {
-        const ci = getShakeScalar(shake);
+        iter++;
+        const ci = computePRNGScalar(prng, seed, (new sjcl.bn(iter)).toBits());
+        // Moved this check out of computePRNGScalar to here
+        if (ci.greaterEquals(CURVE.r)) {
+            i--;
+            continue;
+        }
         const cMi = _scalarMult(ci, chkM[i].point);
         const cZi = _scalarMult(ci, chkZ[i]);
         cM = cM.add(cMi);
         cZ = cZ.add(cZi);
     }
-
     return {M: cM.toAffine(), Z: cZ.toAffine()};
 }
 
 /**
- * Computes an output of seeded SHAKE function as a BigNumber
- * @param {Object} shake seeded SHAKE object
- * @return {sjcl.bn} SHAKE output
+ * Computes an output of a PRNG (using the seed if it is HKDF) as a sjcl bn
+ * object
+ * @param {Object} prng PRNG object for generating output
+ * @param {string} seed hex-encoded seed
+ * @param {sjcl.bitArray} salt optional salt for each PRNG eval
+ * @return {sjcl.bn} PRNG output as scalar value
  */
-function getShakeScalar(shake) {
-    const curveOrder = CURVE.r;
-    const bitLen = sjcl.bitArray.bitLength(curveOrder.toBits());
+function computePRNGScalar(prng, seed, salt) {
+    const bitLen = sjcl.bitArray.bitLength(CURVE.r.toBits());
     const mask = MASK[bitLen % 8];
-    let rnd;
-
-    while (!rnd) {
-        let out = shake.squeeze(32, "hex");
-        // Masking is not strictly necessary for p256 but better to be completely
-        // compatible in case that the curve changes
-        const h = "0x" + out.substr(0, 2);
-        const mh = sjcl.codec.hex.fromBits(sjcl.codec.bytes.toBits([h & mask]));
-        out = mh + out.substr(2);
-        const nOut = getBigNumFromHex(out);
-        // Reject samples outside of correct range
-        if (nOut.greaterEquals(curveOrder)) {
-            continue;
-        }
-        rnd = nOut;
+    let out;
+    switch (prng.name) {
+        case "shake":
+            out = prng.func.squeeze(32, "hex");
+            break;
+        case "hkdf":
+            out = sjcl.codec.hex.fromBits(prng.func(sjcl.codec.hex.toBits(seed), (bitLen+7)/8, sjcl.codec.utf8String.toBits("DLEQ_PROOF"), salt, CURVE_H2C_HASH));
+            break;
+        default:
+            throw new Error(`Server specified PRNG is not compatible: ${prng.name}`);
     }
-    return rnd;
+    // Masking is not strictly necessary for p256 but better to be completely
+    // compatible in case that the curve changes
+    const h = "0x" + out.substr(0, 2);
```

Comment:
Oh, this looked strange at first, but it seems that you want to convert a hex value to a number. That could also be done with `parseInt(out.substr(0, 2), 16)` which would return a Number instead of String.

(The code existed before, it is not really wrong, but mixing types seemed odd.)